File: src/Modelo/Prova.java
package Modelo;

import java.util.Calendar;
import java.util.ArrayList;
import java.text.SimpleDateFormat;

public class Prova extends Atividade{

    private static final long serialVersionUID = 1L;
    ArrayList<String> conteudo;

    public Prova(String nome, Boolean sincrona, Calendar data, ArrayList<String> conteudo){
        super(nome, sincrona);
        this.data = data;
        this.conteudo = conteudo;
    }

    public boolean ehAvaliativa(){
        return true;
    }

    @Override
    public String toString(){
        SimpleDateFormat dt1 = new SimpleDateFormat("dd/MM/yyyy");
        return dt1.format(data.getTime());
    }
}

File: cpp_version/Modelo/Prova.cpp
#include "Prova.h"

Prova :: Prova (string nome, bool sincrona, vector<string> conteudo, time_t data) : Atividade (nome, sincrona){
    this->conteudo = conteudo;
    this->data = data;
}

bool Prova :: ehAvaliativa(){
    return true;
}